include/dem/dem_action_manager.h

```diff
@@ -0,0 +1,429 @@
+  }
+
+  /**
+   * @brief Flag that the grid will motion in the simulation.
+   */
+  inline void
+  set_grid_motion_enabling()
+  {
+    this->grid_motion_enabled = true;
+  }
+
+  /**
+   * @brief Check if the solid objects are enabled to perform some actions.
+   *
+   * @return True if the solid objects are enabled.
+   */
+  inline bool
+  check_grid_motion_enabling()
+  {
+    return grid_motion_enabled;
+  }
+
+  /**
+   * @brief Set triggers for action to be performed in the current time step
+   * because the simulation starts from restart files (previously called
+   * checkpoint step).
+   *
+   * It triggers the reading of checkpoints, the contact search,
+   * clearing the contact structures, the solid object mapping (if enabled), and
+   * the update of the sparse contacts cells (if enabled).
+   */
+  inline void
+  restart_simulation()
+  {
+    read_checkpoint_trigger          = true;
+    contact_search_trigger           = true;
+    clear_tangential_overlap_trigger = true;
+    solid_object_search_trigger      = solid_objects_enabled ? true : false;
+    sparse_contacts_cells_update_trigger =
+      sparse_contacts_enabled ? true : false;
+  }
+
+  /**
+   * @brief Set triggers for action to be performed in the current time step
+   * because of a load balancing step.
+   *
+   * It triggers:
+   * - load balancing: the used method has determined that the load needs to be
+   *                   balanced
+   * - contact search: cells and particles will wont be on the same processor,
+   *                   so all the contact search needs to be performed
+   * - clearing the contact structures: same as above, the contact structures
+   *                                    need to be cleared prior of it
+   * - resizing the containers: the containers need to be resized to the new
+   *                            number of local particles
+   * - solid object search (if enabled): the solid objects need to be mapped to
+   *                                     the rebalanced cells
+   * - update the sparse contacts cells (if enabled): the sparse contacts cells
+   *                                 need to be updated since they only contain
+   *                                 local and ghost cells
+   */
+  inline void
+  load_balance_step()
+  {
+    load_balance_trigger             = true;
+    contact_search_trigger           = true;
+    clear_tangential_overlap_trigger = true;
+    resize_containers_trigger        = true;
+    solid_object_search_trigger      = solid_objects_enabled ? true : false;
+    sparse_contacts_cells_update_trigger =
+      sparse_contacts_enabled ? true : false;
+  }
+
+  /**
+   * @brief Set trigger for the contact search to be performed in the current
+   * time step because of a particle insertion step.
+   *
+   * It triggers:
+   * - contact search: since there are new particles in the simulation, their
+   *                   potential contacts need to be found
+   */
```

Comment:
```suggestion
   * It triggers:
   * - contact search: since there are new particles in the simulation, contact search needs to be performed for all particles
   */
```